### DISCUSSION TITLE
Problem running FINN examples

### DISCUSSION THREAD

**Discussion by Galapple** [Xilinx/finn#999] [Q&A]
I have XRT version 2.11.634 and I am reciving this error while trying to run 1_cifar10_with_cnv_networks, can anyone specify what needs to be done in order to run this notebook?

Below are the cells that I run so far:
=========================================
from finn_examples import models
print(list(filter(lambda x: "cifar10" in x, dir(models))))

accel = models.cnv_w1a1_cifar10()

XRT build version: 2.11.634
Build hash: 5ad5998d67080f00bca5bf15b3838cf35e0a7b26
Build date: 2021-06-09 05:08:58
Git branch: 2021.1
PID: 69674
UID: 1004
[Wed Mar 13 13:50:12 2024 GMT]
HOST: cauchy
EXE: /usr/local/bin/python3.9
[XRT] ERROR: See dmesg log for details. err=-2

---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
Cell In[2], line 1
----> 1 accel = models.cnv_w1a1_cifar10()

File ~/.local/lib/python3.9/site-packages/finn_examples/models.py:274, in cnv_w1a1_cifar10(target_platform)
    272 model_name = "cnv-w1a1"
    273 filename = find_bitfile(model_name, target_platform)
--> 274 return FINNExampleOverlay(filename, driver_mode, _cifar10_cnv_io_shape_dict)

File ~/.local/lib/python3.9/site-packages/finn_examples/driver.py:80, in FINNExampleOverlay.__init__(self, bitfile_name, platform, io_shape_dict, batch_size, fclk_mhz, device, download, runtime_weight_dir)
     48 def __init__(
     49     self,
     50     bitfile_name,
   (...)
     57     runtime_weight_dir="runtime_weights/",
     58 ):
     59     """Initialize the FINN accelerator.
     60 
     61     Parameters
   (...)
     78         Path to runtime weights folder.
     79     """
---> 80     super().__init__(bitfile_name, download=download, device=device)
     81     self.runtime_weight_dir = runtime_weight_dir
     82     self._io_shape_dict = io_shape_dict

File ~/.local/lib/python3.9/site-packages/pynq/overlay.py:348, in Overlay.__init__(self, bitfile_name, dtbo, download, ignore_version, device, gen_cache)
    345     self.systemgraph = None
    347 if download:
--> 348     self.download()
    349 else:
    350     if gen_cache:

File ~/.local/lib/python3.9/site-packages/pynq/overlay.py:428, in Overlay.download(self, dtbo)
    425         else:
    426             Clocks.set_pl_clk(i)
--> 428 super().download(self.parser)
    429 if dtbo:
    430     super().insert_dtbo(dtbo)

File ~/.local/lib/python3.9/site-packages/pynq/bitstream.py:168, in Bitstream.download(self, parser)
    151 def download(self, parser=None):
    152     """Download the bitstream onto PL and update PL information.
    153 
    154     If device tree blob has been specified during initialization, this
   (...)
    166 
    167     """
--> 168     self.device.download(self, parser)

File ~/.local/lib/python3.9/site-packages/pynq/pl_server/xrt_device.py:541, in XrtDevice.download(self, bitstream, parser)
    539 with open(bitstream.bitfile_name, "rb") as f:
    540     data = f.read()
--> 541 self._xrt_download(data)
    542 super().post_download(bitstream, parser)

File ~/.local/lib/python3.9/site-packages/pynq/pl_server/xrt_device.py:529, in XrtDevice._xrt_download(self, data)
    527             xrt.xclOpenContext(self.handle, v["uuid_ctypes"], v["idx"], True)
    528         self.contexts = old_contexts
--> 529         raise RuntimeError(
    530             "Programming Device failed: " + _format_xrt_error(err)
    531         )
    532 finally:
    533     xrt.xclUnlockDevice(self.handle)

RuntimeError: Programming Device failed: EINVAL (22) Invalid argument
===========================================================================
I have also tryed with a super user and it didn't helped.

Thank you

**Comment by auphelia** [comment]
Hi @Galapple, 
Could you please post this issue on finn-examples? 
You can link the created issue in this thread but I am not sure that the finn-examples maintainer screen the finn GitHub Discussions regularly.

**Reply by Galapple** [reply]
Ok I will, thanks.
